tests/test_packages/test_skills/test_tac_control/test_parameters.py

```diff
@@ -0,0 +1,226 @@
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the tests of the parameters module of the tac control skill."""
+
+import datetime
+import logging
+from pathlib import Path
+from typing import cast
+from unittest.mock import Mock, patch
+
+import pytest
+
+from aea.exceptions import AEAEnforceError
+from aea.helpers.search.models import Location
+from aea.test_tools.test_skill import BaseSkillTestCase
+
+from packages.fetchai.skills.tac_control.parameters import Parameters
+
+from tests.conftest import ROOT_DIR
+
+
+class TestParameters(BaseSkillTestCase):
+    """Test Parameters module of tac control."""
+
+    path_to_skill = Path(ROOT_DIR, "packages", "fetchai", "skills", "tac_control")
+
+    @classmethod
+    def setup(cls):
+        """Setup the test class."""
+        super().setup()
+        cls.parameters = cast(Parameters, cls._skill.skill_context.parameters)
+        cls.parameters_manual = Parameters(
+            ledger_id="some_ledger_id",
+            contract_address=None,
+            good_ids=[],
+            currency_ids=[],
+            min_nb_agents=2,
+            money_endowment=200,
+            nb_goods=9,
+            nb_currencies=1,
+            tx_fee=1,
+            base_good_endowment=2,
+            lower_bound_factor=1,
+            upper_bound_factor=1,
+            registration_start_time="01 01 2020  00:01",
+            registration_timeout=60,
+            item_setup_timeout=60,
+            competition_timeout=300,
+            inactivity_timeout=30,
+            whitelist=[],
+            location={"longitude": 51.5194, "latitude": 0.1270},
+            service_data={"key": "tac", "value": "v1"},
+            name="parameters",
+            skill_context=cls._skill.skill_context,
+        )
+
+    def test_init_now_after_start(self):
+        """Test the init of Parameters where now is after the registration_start_time."""
+        mocked_start_time_str = "01 01 2020  00:01"
+        mocked_start_time = datetime.datetime.strptime(
+            mocked_start_time_str, "%d %m %Y %H:%M"
+        )
+        mocked_now_time = datetime.datetime.strptime(
+            "01 01 2020  00:02", "%d %m %Y %H:%M"
+        )
+        datetime_mock = Mock(wraps=datetime.datetime)
+        datetime_mock.now.return_value = mocked_now_time
+
+        with patch("datetime.datetime", new=datetime_mock):
+            with patch.object(self.parameters.context.logger, "log") as mocked_logger:
+                Parameters(
```

Comment:
I'd define kwargs once in setup as a dict, then reuse. Makes it easier to read and see what;'s different across instantiations.

Reply:
Done